app/Http/Controllers/Api/ApiPetController.php

```diff
@@ -0,0 +1,188 @@
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+class ApiPetController extends ApiController
+{
+    /**
+     * Get the detail of a given pet.
+     * @param  Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Request $request, $id)
+    {
+        try {
+            $pet = Pet::where('account_id', auth()->user()->account_id)
+                ->where('id', $id)
+                ->firstOrFail();
+        } catch (ModelNotFoundException $e) {
+            return $this->respondNotFound();
+        }
+
+        return new PetResource($pet);
+    }
+
+    /**
+     * Store the pet.
+     * @param  Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        // Validates basic fields to create the entry
+        $validator = Validator::make($request->all(), [
+            'pet_category_id' => 'integer|required|exists:pet_categories,id',
+            'contact_id' => 'required|integer',
+        ]);
+
+        if ($validator->fails()) {
+            return $this->setErrorCode(32)
+                        ->respondWithError($validator->errors()->all());
+        }
+
+        try {
+            $contact = Contact::where('account_id', auth()->user()->account_id)
+                ->where('id', $request->input('contact_id'))
+                ->firstOrFail();
+        } catch (ModelNotFoundException $e) {
+            return $this->respondNotFound();
+        }
+
+        try {
+            $pet = Pet::create(
+              $request->all()
+              + [
+                'account_id' => auth()->user()->account->id,
+              ]
+            );
+        } catch (QueryException $e) {
+            return $this->respondNotTheRightParameters();
+        }
+
+        return new PetResource($pet);
+    }
+
+    /**
+     * Update the pet.
+     * @param  Request $request
+     * @param  int $petId
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $petId)
+    {
+        try {
+            $pet = Pet::where('account_id', auth()->user()->account_id)
+                ->where('id', $petId)
+                ->firstOrFail();
+        } catch (ModelNotFoundException $e) {
+            return $this->respondNotFound();
+        }
+
+        // Validates basic fields to create the entry
+        $validator = Validator::make($request->all(), [
+            'pet_category_id' => 'sometimes|integer|required|exists:pet_categories,id',
+            'contact_id' => 'sometimes|required|integer',
+        ]);
+
+        if ($validator->fails()) {
+            return $this->setErrorCode(32)
+                        ->respondWithError($validator->errors()->all());
+        }
+
+        try {
+            if ($request->input('contact_id')) {
+                $contact = Contact::where('account_id', auth()->user()->account_id)
+                    ->where('id', $request->input('contact_id'))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "$contact" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1481)